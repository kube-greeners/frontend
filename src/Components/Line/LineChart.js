import React, { useState, useEffect } from 'react';
import { Line } from '@ant-design/charts';
import data from '../../Mockdata/c02emission.json'


/**
 * @param {Array} data - The data to show
 * @param {string} xField - Name of the x axis values
 * @param {string} yField - Name of the y axis values
 */
const LineChart = ({data,xField,yField}) => {


   
    const config = {
        data: data,
        padding: 'auto',
        padding:100,
        xField,
        yField,
        xAxis: {
            tickCount: 10,
            title: {
                text: xField,
                style: { fontSize: 16 },
            },
        },
        yAxis: {
            title: {
                text: yField,
                style: { fontSize: 16 },
            },
        },
        slider: {
            start: 0.9,
            end: 1,
        },
    };
    return <Line {...config} />;
};

export default LineChart;

