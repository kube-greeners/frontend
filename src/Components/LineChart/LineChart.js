import React from "react";
import { Line } from "@ant-design/charts";

/**
 * @param {Array} data - The data to show. Must be an array of objects like
 * @example
 * const data = [
 *      {
 *          valueOne:"test" //X-axis
 *          valueTwo:20 //Y-axis
 *      },
 *      {
 *          valueOne:"test2"
 *          valueTwo:30
 *      }
 * ]
 */
const LineChart = ({ data }) => {
  const [xField, yField] = Object.keys(data[0]);

  const config = {
    data: data,
    xField,
    yField,
    height: 500,
    appendPadding: 20,
    xAxis: {
      tickCount: 10,
      title: {
        text: xField,
        style: { fontSize: 16 },
      },
    },
    yAxis: {
      title: {
        text: yField,
        style: { fontSize: 16 },
      },
    },
    slider: {
      start: 0,
      end: 0.99,
      trendCfg: {
        lineStyle: {
          opacity: 0.1,
          stroke: "black",
        },
      },
    },
  };
  return <Line {...config} />;
};

export default LineChart;
