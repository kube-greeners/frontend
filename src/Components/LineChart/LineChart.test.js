import LineChart from "./LineChart";
import renderer from "react-test-renderer";

test("matches LineChart snapshot", () => {
    const data = [
        {
            valueOne: "test", //X-axis
            valueTwo: 20 //Y-axis
        },
        {
            valueOne: "test2",
            valueTwo: 30
        }
    ]
    const tree = renderer.create(<LineChart data={data}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
