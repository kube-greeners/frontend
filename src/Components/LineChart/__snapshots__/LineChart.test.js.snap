// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches LineChart snapshot 1`] = `
<div
  style={
    Object {
      "height": "inherit",
    }
  }
/>
`;
